src/thermo/PengRobinson.cpp

```diff
@@ -0,0 +1,773 @@
+    // based on crit properties T_c and P_c:
+    for (size_t isp = 0; isp < nDatabase; isp++) {
+        XML_Node& acNodeDoc = doc->child(isp);
+        std::string iNameLower = toLowerCopy(iName);
+        std::string dbName = toLowerCopy(acNodeDoc.attrib("name"));
+
+        // Attempt to match provided species iName to current database species
+        //  dbName:
+        if (iNameLower == dbName) {
+            // Read from database and calculate a and b coefficients
+            double vParams;
+            double T_crit = 0.0, P_crit = 0.0, w_ac = 0.0;
+
+            if (acNodeDoc.hasChild("Tc")) {
+                vParams = 0.0;
+                XML_Node& xmlChildCoeff = acNodeDoc.child("Tc");
+                if (xmlChildCoeff.hasAttrib("value")) {
+                    std::string critTemp = xmlChildCoeff.attrib("value");
+                    vParams = strSItoDbl(critTemp);
+                }
+                if (vParams <= 0.0) { //Assuming that Pc and Tc are non zero.
+                    throw CanteraError("PengRobinson::getCoeff",
+                        "Critical Temperature must be positive");
+                }
+                T_crit = vParams;
+            }
+            if (acNodeDoc.hasChild("Pc")) {
+                vParams = 0.0;
+                XML_Node& xmlChildCoeff = acNodeDoc.child("Pc");
+                if (xmlChildCoeff.hasAttrib("value")) {
+                    std::string critPressure = xmlChildCoeff.attrib("value");
+                    vParams = strSItoDbl(critPressure);
+                }
+                if (vParams <= 0.0) { //Assuming that Pc and Tc are non zero.
+                    throw CanteraError("PengRobinson::getCoeff",
+                        "Critical Pressure must be positive");
+                }
+                P_crit = vParams;
+            }
+            if (acNodeDoc.hasChild("omega")) {
+                vParams = 0.0;
+                XML_Node& xmlChildCoeff = acNodeDoc.child("omega");
+                if (xmlChildCoeff.hasChild("value")) {
+                    std::string acentric_factor = xmlChildCoeff.attrib("value");
+                    vParams = strSItoDbl(acentric_factor);
+                }
+                w_ac = vParams;
+            }
+
+            spCoeff[0] = omega_a * (GasConstant * GasConstant) * (T_crit * T_crit) / P_crit; //coeff a
+            spCoeff[1] = omega_b * GasConstant * T_crit / P_crit; // coeff b
+            spCoeff[2] = w_ac; // acentric factor
+            break;
+        }
+    }
+    // If the species is not present in the database, throw an error 
+    if(isnan(spCoeff[0]))
+    {
+        throw CanteraError("PengRobinson::getCoeff",
+            "Species '{}' is not present in the database", iName);
+    }
+    return spCoeff;
+}
+
+void PengRobinson::initThermo()
+{
+    for (auto& item : m_species) {
+        // Read a and b coefficients and acentric factor w_ac from species 'input'
+        // information (i.e. as specified in a YAML input file)
```

Comment:
Why is `'input'` in quotes here? We're also trying to avoid the use of Latin abbreviations such as `i.e.` in new code, since they may not be familiar to everyone.

Reply:
Corrected.